src/app/clientSync/ClientSync.ts

```diff
@@ -0,0 +1,112 @@
+import RuntimeError from "../errors/RuntimeError";
+import ValidationError from "../errors/ValidationError";
+import { ClaimRequest } from "../requests/ClaimRequest";
+import { PublishRequest } from "../requests/PublishRequest";
+import { BountyCollection } from "../types/bounty/BountyCollection";
+import { ChangeStreamEvent } from "../types/mongo/ChangeStream";
+import Log, { LogUtils } from "../utils/Log";
+
+/**
+ * Handles some basic validation/safety logic to prevent unintentional double calls to activities.
+ * For example, most bounty lifecycle activities require writing to the bounty record the channel id
+ * and message id of the bounty card (whether in a Text or DM Channel). In that case, we don't want to kick
+ * off the activity that generated that db operation.
+ * The other case is that a user kicks off an activity on the bot, like claim. 
+ * This will result in a write operation to the db, and we don't want to kick off the claim activity again. 
+ * @param args an object containing the ChangeStreamEvent
+ * @returns void promise
+ */
+export const ClientSync = async (args: {changeStreamEvent: ChangeStreamEvent}): Promise<void> => {
+    // if OperationType is insert, updatedFields will be invalid
+    let filterEvent = (
+        "update" === args.changeStreamEvent.operationType && 
+        !args.changeStreamEvent.updateDescription.updatedFields.activityHistory
+    );
+    
+    if (filterEvent) {
+        console.log(`filtered type of event ${args.changeStreamEvent.operationType} with no activity history`);
+        return;
+    }
+    
+    if (args.changeStreamEvent.fullDocument) {
+        const activityHistory = args.changeStreamEvent.fullDocument.activityHistory;
+        console.log('here');
+        if (activityHistory && activityHistory[activityHistory.length - 1].client != Clients.bountybot) {
+            changeStreamEventHandler(args.changeStreamEvent);
+        } else {
+            // no-op: don't process bot changes to the db
+            console.log('no activity history');
+        }
+    }
+}
+
+/**
+ * Handles the object passed from a mongodb changestream listener.
+ * Transforms changeStreamEvent to a *Request object that can be processed by /bounty activity classes.
+ * @param changeStreamEvent
+ *     The object passed from the mongodb changestream listener
+ */
+const changeStreamEventHandler= async (event: ChangeStreamEvent): Promise<void> => {
+    let request: any;
+    const interactionHistory = event.fullDocument.activityHistory;
+    const lastClientActivity = interactionHistory[interactionHistory.length - 1];
+    const activity = lastClientActivity.activity;
+    Log.info(`Processing ${activity} activity event. Origination: ${Clients.bountyboardweb}`);
+    switch (activity) {
+        case Activities.create:
+            // no-op
+            break;
+        case Activities.publish:
+            Log.info('verify new bounty received');
+            // TODO: add field to front end
```

Comment:
Which field?

Reply:
`requireApplication`,  for multi-apply/pitch bounties